python/cudf/cudf/core/index.py

```diff
@@ -1641,10 +1832,106 @@ def as_index(arbitrary, **kwargs):
     )
 
 
-def _setdefault_name(values, kwargs):
-    if "name" not in kwargs:
+def _setdefault_name(values, **kwargs):
+    if "name" not in kwargs or kwargs["name"] is None:
         if not hasattr(values, "name"):
-            kwargs.setdefault("name", None)
+            kwargs.update({"name": None})
         else:
-            kwargs.setdefault("name", values.name)
+            kwargs.update({"name": values.name})
     return kwargs
+
+
+def Int64Index(data=None, dtype="int64", copy=False, name=None):
+    """Immutable ndarray implementing an ordered, sliceable set.
+    The basic object storing row labels for all cuDF objects.
+    Int64Index is a special case of Index with purely integer labels.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype (default: int64)
+            If dtype is None, we find the dtype that best fits the data.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+
+    Examples
+    --------
+    >>> import cudf
+    >>> cudf.Int64Index([1, 2, 3, 4], name="a")
+    Int64Index([1, 2, 3, 4], dtype='int64', name='a')
+
+    >>> cudf.Int64Index([1, 2, 3, 4], dtype="int16", name="a")
+    Int64Index([1, 2, 3, 4], dtype='int64', name='a')
+    """
+
+    return Index(data=data, dtype=dtype, copy=copy, name=name)
```

Comment:
`Int64Index` should be a class where ideally someone can do something like `isinstance(obj, Int64Index)`

Reply:
Any ideas on how we could handle this?

Reply:
It would be better to create a new class as we need to have different default values for different index types.